### PR TITLE
fix cli: vela show support list the parameter of ComponentDefinition which use helm charts

### PR DESCRIPTION
**Pull request by yangsoon**
fix https://github.com/oam-dev/kubevela/issues/1394

vela show now support list the parameter of ComponentDefinition which use helm charts as workload. 

When a new ComponentDefinition use helm charts is registered,  the ComponentDefinition controller will parse
the values.yaml in charts and store as openapi into configMap. (refer to https://github.com/oam-dev/kubevela/pull/1250)

vela show will get the openapi data in configMap, and list the parameters to the terminal.

take [webapp-chart](https://github.com/oam-dev/kubevela/blob/master/docs/examples/helm-module/webapp-chart-cd.yaml) as an example.

``` 
$ vela show webapp-chart
# Properties
+----------------+-------------------------------------------+-----------------------------------+----------+---------+
|      NAME      |                DESCRIPTION                |               TYPE                | REQUIRED | DEFAULT |
+----------------+-------------------------------------------+-----------------------------------+----------+---------+
| backends       |                                           | array                             | false    | []      |
| faults         |                                           | [faults](#faults)                 | false    | {}      |
| linkerd        |                                           | [linkerd](#linkerd)               | false    | {}      |
| serviceMonitor |                                           | [serviceMonitor](#serviceMonitor) | false    | {}      |
| ui             |                                           | [ui](#ui)                         | false    | {}      |
| podAnnotations |                                           | [podAnnotations](#podAnnotations) | false    | {}      |
| resources      |                                           | [resources](#resources)           | false    | {}      |
| tolerations    |                                           | array                             | false    | []      |
| affinity       |                                           | [affinity](#affinity)             | false    | {}      |
| backend        |                                           |                                   | false    |         |
| certificate    | create a certificate manager certificate  | [certificate](#certificate)       | false    | {}      |
| image          |                                           | [image](#image)                   | false    | {}      |
| logLevel       |                                           | string                            | false    | info    |
| redis          | Redis deployment                          | [redis](#redis)                   | false    | {}      |
| serviceAccount |                                           | [serviceAccount](#serviceAccount) | false    | {}      |
| cache          | Redis address in the format <host>:<port> | string                            | false    |         |
| h2c            |                                           | [h2c](#h2c)                       | false    | {}      |
| hpa            | metrics-server add-on required            | [hpa](#hpa)                       | false    | {}      |
| ingress        |                                           | [ingress](#ingress)               | false    | {}      |
| nodeSelector   |                                           | [nodeSelector](#nodeSelector)     | false    | {}      |
| replicaCount   |                                           | integer                           | false    |       1 |
| service        |                                           | [service](#service)               | false    | {}      |
| tls            | enable tls on the podinfo service         | [tls](#tls)                       | false    | {}      |
+----------------+-------------------------------------------+-----------------------------------+----------+---------+


## faults
+-------------+-------------+---------+----------+---------+
|    NAME     | DESCRIPTION |  TYPE   | REQUIRED | DEFAULT |
+-------------+-------------+---------+----------+---------+
| testFail    |             | boolean | false    | false   |
| testTimeout |             | boolean | false    | false   |
| unhealthy   |             | boolean | false    | false   |
| unready     |             | boolean | false    | false   |
| delay       |             | boolean | false    | false   |
| error       |             | boolean | false    | false   |
+-------------+-------------+---------+----------+---------+


## linkerd
+---------+-------------+---------------------+----------+---------+
|  NAME   | DESCRIPTION |        TYPE         | REQUIRED | DEFAULT |
+---------+-------------+---------------------+----------+---------+
| profile |             | [profile](#profile) | false    | {}      |
+---------+-------------+---------------------+----------+---------+


### profile
+---------+-------------+---------+----------+---------+
|  NAME   | DESCRIPTION |  TYPE   | REQUIRED | DEFAULT |
+---------+-------------+---------+----------+---------+
| enabled |             | boolean | false    | false   |
+---------+-------------+---------+----------+---------+


## serviceMonitor
+----------+-------------+---------+----------+---------+
|   NAME   | DESCRIPTION |  TYPE   | REQUIRED | DEFAULT |
+----------+-------------+---------+----------+---------+
| enabled  |             | boolean | false    | false   |
| interval |             | string  | false    | 15s     |
+----------+-------------+---------+----------+---------+


## ui
+---------+-------------+--------+----------+---------+
|  NAME   | DESCRIPTION |  TYPE  | REQUIRED | DEFAULT |
+---------+-------------+--------+----------+---------+
| color   |             | string | false    | #34577c |
| logo    |             | string | false    |         |
| message |             | string | false    |         |
+---------+-------------+--------+----------+---------+


## resources
+----------+-------------+-----------------------+----------+---------+
|   NAME   | DESCRIPTION |         TYPE          | REQUIRED | DEFAULT |
+----------+-------------+-----------------------+----------+---------+
| limits   |             |                       | false    |         |
| requests |             | [requests](#requests) | false    | {}      |
+----------+-------------+-----------------------+----------+---------+


### requests
+--------+-------------+--------+----------+---------+
|  NAME  | DESCRIPTION |  TYPE  | REQUIRED | DEFAULT |
+--------+-------------+--------+----------+---------+
| cpu    |             | string | false    | 1m      |
| memory |             | string | false    | 16Mi    |
+--------+-------------+--------+----------+---------+


## certificate
+-----------+--------------------------------------------------------------+-------------------------+----------+-----------+
|   NAME    |                         DESCRIPTION                          |          TYPE           | REQUIRED |  DEFAULT  |
+-----------+--------------------------------------------------------------+-------------------------+----------+-----------+
| create    |                                                              | boolean                 | false    | false     |
| dnsNames  | the hostname / subject alternative names for the certificate | array                   | false    | [podinfo] |
| issuerRef | the issuer used to issue the certificate                     | [issuerRef](#issuerRef) | false    | {}        |
+-----------+--------------------------------------------------------------+-------------------------+----------+-----------+


### issuerRef
+------+-------------+--------+----------+---------------+
| NAME | DESCRIPTION |  TYPE  | REQUIRED |    DEFAULT    |
+------+-------------+--------+----------+---------------+
| kind |             | string | false    | ClusterIssuer |
| name |             | string | false    | self-signed   |
+------+-------------+--------+----------+---------------+

.......
```

```
vela show webapp-chart --web
```

截屏2021-04-22 上午11 24 17

### CODE REVIEW

**Comment by zzxwill** [comment]
You can add `--web` to display the reference doc, and maybe you may want to refine `Type` column and table name.

**Comment by yangsoon** [comment]
> You can add `--web` to display the reference doc, and maybe you may want to refine `Type` column and table name.

截屏2021-04-21 下午5 15 47

I just implemented the feature to display in web

**Comment by zzxwill** [comment]
> > You can add `--web` to display the reference doc, and maybe you may want to refine `Type` column and table name.
> 
> 
> I just implemented the feature to display in web, please \cc @zzxwill

Great. But now you can see it's difficult to explorer the fields like `service`.

**Comment by wonderflow** [comment]
@zzxwill What's the original behaviar?

File: references/plugins/references.go [hunk 1]
```diff
@@ -446,3 +512,91 @@ func (ref *ConsoleReference) GenerateCapabilityProperties(capability *types.Capa
...
+
+	var consoleRefs []ConsoleReference
+	for _, item := range helmRefs {
+		consoleRefs = append(consoleRefs, ref.prepareParameter(item.Name, item.Parameters, types.HelmCategory))
+	}
+	return helmRefs, consoleRefs, err
+}
+
+// WalkParameterSchema will extract properties from *openapi3.Schema
+func WalkParameterSchema(parameters *openapi3.Schema, name string, depth int) {
```
Comment (wonderflow) [inline references/plugins/references.go:557]: write a test for it

Comment (yangsoon) [reply references/plugins/references.go:557]: got it

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**